src/sas/qtgui/Plotting/Slicers/WedgeSlicer.py

```diff
@@ -211,38 +211,41 @@ def _post_data(self, new_sector=None, nbins=None):
 
     def validate(self, param_name, param_value):
         """
-        Validate input from user
+        Validate input from user.
         Values get checked at apply time.
         """
-        MIN_Q_DIFFERENCE = self.dqmin
-        MIN_PHI_DIFFERENCE = 0.01
-        isValid = True
-
-        # Stitched together from other slicers. There could be a neater way.
-        if param_name == 'r_min':
-            if np.fabs(param_value - self.getParams()['r_max']) < MIN_Q_DIFFERENCE:
-                print("Inner and outer radii too close. Please adjust.")
-                isValid = False
-            elif param_value > self.qmax:
-                print("Inner radius exceeds maximum range. Please adjust.")
-                isValid = False
-        elif param_name == 'r_max':
-            if np.fabs(param_value - self.getParams()['r_min']) < MIN_Q_DIFFERENCE:
-                print("Inner and outer radii too close. Please adjust.")
-                isValid = False
+
+        def check_radius_difference(param_name, other_radius_name, param_value):
+            if np.fabs(param_value - self.getParams()[other_radius_name]) < self.dqmin:
+                return "Inner and outer radii too close. Please adjust."
             elif param_value > self.qmax:
-                print("Outer radius exceeds maximum range. Please adjust.")
-                isValid = False
-        elif param_name == 'delta_phi [deg]':
-            if np.fabs(param_value) < MIN_PHI_DIFFERENCE:
-                print("Sector angles too close. Please adjust.")
-                isValid = False
-        elif param_name == 'nbins':
-            # Can't be 0
+                return f"{param_name} exceeds maximum range. Please adjust."
+            return None
+
+        def check_phi_difference(param_value):
+            if np.fabs(param_value) < 0.01:
+                return "Sector angles too close. Please adjust."
+            return None
+
+        def check_bins(param_value):
             if param_value < 1:
-                print("Number of bins cannot be <= 0. Please adjust.")
-                isValid = False
-        return isValid
+                return "Number of bins cannot be <= 0. Please adjust."
+            return None
+
+        validators = {
+            'r_min': lambda value: check_radius_difference('r_min', 'r_max', value),
+            'r_max': lambda value: check_radius_difference('r_max', 'r_min', value),
+            'delta_phi [deg]': check_phi_difference,
```

Comment:
setting delta_phi very low in windows gui (e.g. 0.000001 deg) does not give any error message in the log, but still the plot window does not react, so an error message should be created!

Reply:
On a Linux host I am getting error messages for very small delta_phi angles, either "ValueError: Average Error: No points inside ROI to average..." or "Sector angles too close. Please adjust." depending on where I've placed the slicer. Now that I'm no longer employed by STFC/RAL/ISIS, I don't have access to a windows development machine. Is anyone else able to help with this?

Reply:
I have tried in the main branch (Windows GUI) as well as in the #2570 branch (Windows GUI) 
--> for varying phi/delta_phi: yes, in both it completely depends which phi and which delta_phi together with which nbins I choose. The higher nbins, the smaller delta_phi can be (also <1 and the plot still works)
--> for varying r_min: I can choose at as small as I want (eg 0.00000001) and no error message pops up.
We can hopefully speak about this later in the code camp.